app/components/ToggleDisplay.js

```diff
@@ -80,28 +79,17 @@ export default class ToggleDisplay extends Component {
 
   runHelpers = () => {
     if (this.state.helpers) {
-      let shouldHide = false;
-      let obj = {};
-
       chrome.storage.local.get('display', result => {
-        this.state.helpers.map(helper => {
+        this.state.helpers.forEach(helper => {
           if (helper.type === 'hide') {
             if (!{}.hasOwnProperty.call(result.display, helper.relation)) {
-              return helper;
+              return;
             }
             const relation = result.display[helper.relation].visible;
-            shouldHide = this.state.visible + relation;
-            obj = helper;
-            obj.display = true;
+            helper.display = (this.state.visible + relation) !== 0;
+            this.sendHelperToDOM(helper);
           }
```

Comment:
Okay cool, just wanted to make sure we refactored to what we wanted it to be.

I'll merge it in!